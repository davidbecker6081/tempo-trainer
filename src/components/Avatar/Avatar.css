.avatar {
  padding: 1rem;
  margin-bottom: 2rem;
  width: 85%;
  border-bottom: 1px solid #4ee46f40;
  letter-spacing: 1.1px;
}

.avatar img {
  height: 10rem;
  width: 10rem;
  border: 2px solid #4ee46f40;
  border-radius: 50%;
  opacity: 0.9;
}

.avatar img:hover,
.avatar div:hover {
  box-shadow: 0px 0px 2px 2px #4ee46f2b;
  cursor: pointer;
  opacity: 1;
}

@media screen and (max-width: 1150px) {
  .avatar img {
    height: 7rem;
    width: 7rem;
  }

  .avatar {
    width: 80%;
  }
}

@media screen and (max-width: 770px) {

}
