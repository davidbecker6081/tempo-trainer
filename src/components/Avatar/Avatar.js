import React from 'react';
import PropTypes from 'prop-types';
import './Avatar.css';

const Avatar = ({ user }) => {
  const name = user.name;
  const userImage = user.profileImg;

  return (
    <article className="avatar-container">
      <img src={userImage} alt="user profile" />
      <h1>{name}</h1>
    </article>
  );
};

export default Avatar;

Avatar.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
    profileImg: PropTypes.string.isRequired,
  }).isRequired,
};
