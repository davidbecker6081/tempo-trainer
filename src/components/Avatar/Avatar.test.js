import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Avatar from './Avatar';

describe('Avatar', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      user: {
        name: '',
        userImage: '',
      },
    };
  });

  describe('name is defined', () => {
    beforeEach(() => { mockProps.name = 'David'; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('userImage is defined', () => {
    beforeEach(() => { mockProps.userImage = 'test url'; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
