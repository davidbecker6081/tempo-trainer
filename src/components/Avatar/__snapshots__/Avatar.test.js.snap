// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar name is defined should render correclty 1`] = `undefined`;

exports[`Avatar userImage is defined should render correclty 1`] = `undefined`;
